build-tools/packages/build-cli/src/handlers/promptFunctions.ts

```diff
@@ -0,0 +1,296 @@
+ * @param log - A logger that the function can use for logging.
+ * @param data - An object with handler-specific contextual data.
+ * @returns True if the state was handled; false otherwise.
+ */
+export const promptToRelease: StateHandlerFunction = async (
+    state: MachineState,
+    machine: Machine<unknown>,
+    testMode: boolean,
+    log: CommandLogger,
+    data: FluidReleaseStateHandlerData,
+): Promise<boolean> => {
+    if (testMode) return true;
+
+    const { command, context, releaseGroup, releaseVersion, promptWriter } = data;
+    assert(context !== undefined, "Context is undefined.");
+
+    const flag = isReleaseGroup(releaseGroup) ? "-g" : "-p";
+    const prompt: InstructionalPrompt = {
+        title: `READY TO RELEASE version ${chalk.bold(releaseVersion!)}!`,
+        sections: [
+            {
+                title: "FIRST",
+                message: `Queue a ${chalk.green(
+                    chalk.bold("release"),
+                )} build for the following release group in ADO for branch ${chalk.blue(
+                    chalk.bold(context.originalBranchName),
+                )}:\n\n    ${chalk.green(chalk.bold(releaseGroup!))}`,
+            },
+            {
+                title: "NEXT",
+                message: `After the build is done and the release group has been published, run the following command to bump the release group to the next version and update dependencies on the newly released package(s):`,
+                cmd: `${command?.config.bin} ${command?.id} ${flag} ${releaseGroup}`,
+            },
+        ],
+    };
+
+    await promptWriter?.writePrompt(prompt);
+    return true;
+};
+
+/**
+ * Prompt the user to queue a release build.
+ *
+ * @param state - The current state machine state.
+ * @param machine - The state machine.
+ * @param testMode - Set to true to run function in test mode.
+ * @param log - A logger that the function can use for logging.
+ * @param data - An object with handler-specific contextual data.
+ * @returns True if the state was handled; false otherwise.
+ */
+export const promptToReleaseDeps: StateHandlerFunction = async (
+    state: MachineState,
+    machine: Machine<unknown>,
+    testMode: boolean,
+    log: CommandLogger,
+    data: FluidReleaseStateHandlerData,
+): Promise<boolean> => {
+    if (testMode) return true;
+
+    const { context, promptWriter, releaseGroup } = data;
+    assert(context !== undefined, "Context is undefined.");
+    assert(promptWriter !== undefined, "promptWriter is undefined.");
+
+    const prereleaseDepNames = await getPreReleaseDependencies(context, releaseGroup!);
+
+    const prompt: InstructionalPrompt = {
+        title: "NEED TO RELEASE DEPENDENCIES",
+        sections: [
+            {
+                title: "DETAILS",
+                message: chalk.red(
+                    `\nCan't release the ${releaseGroup} release group because some of its dependencies need to be released first.`,
```

Comment:
```suggestion
                    `Can't release the ${releaseGroup} release group because some of its dependencies need to be released first.`,
```